### PR TITLE
puller(ticdc): remove span hack func call from frontier Forward

### LINKED ISSUE #7202: remove span hack func call from frontier Forward
remove span hack func call from frontier Forward

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

Issue Number: close #7202

### What is changed and how it works?
the End key return by the PD API will be nil to represent the biggest key,  now we will try to Hanck the EndKey every time when forwarding resolvedTs,  but it's better to Hack when span is created.

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: cdc/puller/frontier/frontier.go [hunk 1]
```diff
@@ -68,7 +68,6 @@ func (s *spanFrontier) Frontier() uint64 {
 
 // Forward advances the timestamp for a span.
 func (s *spanFrontier) Forward(span regionspan.ComparableSpan, ts uint64) {
-	span = span.Hack()
```
Comment (asddongmen) [inline cdc/puller/frontier/frontier.go:71]: Is it possible to remove `span.Hack()` at https://github.com/pingcap/tiflow/pull/7203/files#diff-01b25d9f5901400302bd5f2b44a7f5e754b0037fd5fcec92d2a7acf066d25acaL39 too? It is little bit confused.

**Review by asddongmen — APPROVED** [review]

**Review by 3AceShowHand — APPROVED** [review]

**Pull request merged**